cdap-data-fabric-tests/src/test/java/co/cask/cdap/data2/transaction/stream/StreamAdminTest.java

```diff
@@ -74,19 +111,142 @@ public void testCreateExist() throws Exception {
+      Assert.fail("User should not be able to create a stream in this namespace.");
+    } catch (UnauthorizedException e) {
+      // expected
+    }
+
+    // grant write access for user to foo_namespace
+    grantAndAssertSuccess(streamId.getNamespace().toEntityId(), USER, ImmutableSet.of(Action.WRITE));
     streamAdmin.create(streamId);
+
     // Even though both streams have the same name, {@code otherStreamId} does not exist because it is in a different
     // namespace than the one created above.
     Assert.assertTrue(streamAdmin.exists(streamId));
     Assert.assertFalse(streamAdmin.exists(otherStreamId));
 
+    try {
+      streamAdmin.create(otherStreamId);
+      Assert.fail("User should not be able to create a stream in this namespace.");
+    } catch (UnauthorizedException e) {
+      // expected
+    }
+
+    // grant write access for user to other_namespace
+    grantAndAssertSuccess(otherStreamId.getNamespace().toEntityId(), USER, ImmutableSet.of(Action.WRITE));
     streamAdmin.create(otherStreamId);
     Assert.assertTrue(streamAdmin.exists(otherStreamId));
+
+    // the user should be able to drop the stream, they had created
+    streamAdmin.drop(otherStreamId);
+    Assert.assertFalse(streamAdmin.exists(otherStreamId));
+
+    // revoke the permission for the user on the stream in foo_namespace
+    revokeAndAssertSuccess(streamId.toEntityId(), USER, ImmutableSet.of(Action.ADMIN, Action.ALL));
+
+    try {
+      streamAdmin.drop(streamId);
+      Assert.fail("User should not be able to delete a stream in this namespace.");
+    } catch (UnauthorizedException e) {
+      // expected
+    }
+
+    // grant WRITE permission to the user but they still should not be able to drop the stream
+    grantAndAssertSuccess(streamId.toEntityId(), USER, ImmutableSet.of(Action.WRITE));
+
+    try {
+      streamAdmin.drop(streamId);
+      Assert.fail("User should not be able to delete a stream with only Write Action access.");
+    } catch (UnauthorizedException e) {
+      // expected
+    }
+
+    // grant admin access and the user should then be able to drop the stream
+    grantAndAssertSuccess(streamId.toEntityId(), USER, ImmutableSet.of(Action.ADMIN));
+    streamAdmin.drop(streamId);
+    Assert.assertFalse(streamAdmin.exists(streamId));
+  }
+
+  @Test
+  public void testConfigAndTruncate() throws Exception {
+    StreamAdmin streamAdmin = getStreamAdmin();
+    grantAndAssertSuccess(new NamespaceId(FOO_NAMESPACE), USER, ImmutableSet.of(Action.WRITE));
+    Id.Stream stream = Id.Stream.from(FOO_NAMESPACE, "stream");
+
+    streamAdmin.create(stream);
+    Assert.assertTrue(streamAdmin.exists(stream));
+    writeEvent(stream);
+
+    // Getting config / properties should work
+    streamAdmin.getConfig(stream);
+    streamAdmin.getProperties(stream);
+
+    // Now revoke access to the user to the stream
+    revokeAndAssertSuccess(stream.toEntityId(), USER, ImmutableSet.of(Action.ALL));
+    streamAdmin.getConfig(stream);
```

Comment:
getConfig doesn't check for authorization. This is required since it is invoked in multiple different places in programs, upgrade tool and in other places. If we need authorization we need to add it before we call getConfig.